### ISSUE TITLE
Document workaround for packagist IPv6 issues on broken local setups

### ISSUE THREAD

**Issue by naderman** [composer/composer#4748]
See https://github.com/composer/composer/issues/4142#issuecomment-117648869 for what needs to be added to the docs.

**Comment by VANSMOOKEYMAWA** [comment]
Leave a comment.good

**Comment by cgrossde** [comment]
Summary of what needs to be documented:


**Workaround Linux:**

On linux, it seems that running this command helps to make ipv4 traffic have a higher prio than ipv6, which is a better alternative than disabling ipv6 entirely:

```Bash
sudo sh -c "echo 'precedence ::ffff:0:0/96 100' >> /etc/gai.conf"
```

**Workaround Windows:**

On windows the only way is to disable ipv6 entirely I am afraid (either in windows or in your home router).

**Workaround Mac OS X:**

Get name of your network device:

```
networksetup -listallnetworkservices
```

Disable IPv6 on that device (in my case "Wi-Fi")

```
networksetup -setv6off Wi-Fi
```

Do your composer thingy ....

You can enable IPv6 again with:

```
networksetup -setv6automatic Wi-Fi
```

That said, if this fixes your problem, please talk to your ISP about it to try and resolve the routing errors. That's the best way to get things resolved for everyone.

**Comment by cgrossde** [comment]
See PR: #4842

**Comment by nat-mystudiosessions** [comment]
I'm on Windows. My composer install was moving crazy slow on one of my workstations but not the others. I troubleshooted this for a long time. The best clue was I discovered for some reason my computer wouldn't ping IPv6 addresses on the command line. The only solution derived from @cgrossde suggestion. I disabled IPv6 in my network card setting, voila, composer started moving quickly again.

What's puzzling to me is my other workstations have IPv6 checked and they work fine, so I can't honestly say that's the real problem. For now, I'm just glad to be up and running. I hope this helps someone else. Hopefully a better fix will emerge.
